### PR TITLE
Add form for create goals

### PR DESCRIPTION
**Pull request by sahsisunny**
### What is the change?

**Following files are added :**

1. `website/components/GoalFormModal/GoalFormModal.vue` : This is the component for the modal for adding a goal.
2. `website/components/goal-form/GoalForm.vue` : This file contains the form for adding a goal.

**Following files are modified :**

1. `website/.eslintrc.js` : Modified the eslint config file to fix the linting errors.
2. `website/pages/index.vue` : Add the Goal Form Modal button.
3. `website/package.json` : Added the vuetify package.
4. `website/yarn.lock` : Added the vuetify package lock file.

**Packages added :**

1. `vuetify@2.6.14` : This is the package for the UI components.

### Is Development Tested?

- [ ] Yes
- [ ] No

### After Change Screenshots
image

### CODE REVIEW

**Comment by pallabez** [comment]
Nice work with the form. There would be some changes as to what base component we will use. Let's discuss on this.

File: website/components/GoalForm/GoalForm.vue [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+    ReusableButton
+  },
+  data: () => ({
+    usernameCategoryOptions: ['Select a user role', 'MEMBER', 'ADMIN', 'DEVELOPER', 'TESTER', 'NEWCOMER', 'DESIGNER'],
+    goalPriorityOptions: ['Select the priority', 'HIGH', 'MEDIUM', 'LOW'],
+    goalTypeOptions: ['Select the type', 'BUG', 'FEATURE', 'DOCUMENTATION', 'TESTING']
+  })
+}
+</script>
+<style>
```
Comment (pallabez) [inline website/components/GoalForm/GoalForm.vue:109]: Can we use scoped?

File: website/components/GoalForm/GoalForm.vue [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+        />
+      </div>
+      <ReusableButton
+        id="submit"
+        type="submit"
+        name="submit"
+        label="Submit"
+      />
+    </form>
+  </div>
```
Comment (pallabez) [inline website/components/GoalForm/GoalForm.vue:87]: Can we use Grid layout component?

**Comment by pallabez** [comment]
Can we remove the Reusable/ components?

**Comment by pallabez** [comment]
Can we add a field for description?

File: website/components/GoalForm/GoalForm.vue [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+          placeholder="eg: 2020-01-01"
+        />
+        <InputField
+          id="assignee"
+          label="Assignee"
+          type="text"
+          name="assignee"
+          required
+          placeholder="eg: ankush"
+        />
```
Comment (pallabez) [inline website/components/GoalForm/GoalForm.vue:78]: Can we use the use the `PopupAssignee` component here?

File: website/components/GoalForm/GoalForm.vue [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+          name="priority"
+          required
+          :options="goalPriorityOptions"
+        />
+        <SelectOption
+          label="Type"
+          name="type"
+          required
+          :options="goalTypeOptions"
+        />
```
Comment (pallabez) [inline website/components/GoalForm/GoalForm.vue:60]: We don't need this for now.

File: website/components/GoalForm/GoalForm.vue [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+          placeholder="eg: sahsisunny"
+        />
+      </div>
+      <div class="input-container">
+        <SelectOption
+          label="Priority"
+          name="priority"
+          required
+          :options="goalPriorityOptions"
+        />
```
Comment (pallabez) [inline website/components/GoalForm/GoalForm.vue:54]: We don't need this for now

File: website/components/GoalForm/GoalForm.vue [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+        required
+        placeholder="Comma separated links"
+      />
+      <div class="input-container">
+        <SelectOption
+          label="User Role"
+          name="user-role"
+          required
+          :options="usernameCategoryOptions"
+        />
```
Comment (pallabez) [inline website/components/GoalForm/GoalForm.vue:38]: This field will be automatically provided by backend, so we won't need it

File: website/components/GoalForm/GoalForm.vue [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+        <InputField
+          id="username"
+          label="Username"
+          type="text"
+          name="username"
+          required
+          placeholder="eg: sahsisunny"
+        />
```
Comment (pallabez) [inline website/components/GoalForm/GoalForm.vue:46]: We won't need this as the data is provided by PopupAssignee

File: website/components/GoalForm/GoalForm.vue [hunk 8]
```diff
@@ -0,0 +1,128 @@
...
+        placeholder="eg: Testing setup for dashboard"
+      />
+      <InputField
+        id="links"
+        label="Links"
+        type="text"
+        name="links"
+        required
+        placeholder="Comma separated links"
+      />
```
Comment (pallabez) [inline website/components/GoalForm/GoalForm.vue:31]: We won't this need as of now.

File: website/components/GoalForm/GoalForm.vue [hunk 9]
```diff
@@ -0,0 +1,128 @@
...
+        autofocus
+      />
+      <InputField
+        id="title"
+        label="Task Name"
+        type="text"
+        name="title"
+        required
+        placeholder="eg: Testing setup for dashboard"
+      />
```
Comment (pallabez) [inline website/components/GoalForm/GoalForm.vue:23]: We won't need this for now.

File: website/components/GoalForm/GoalForm.vue (hunk 3 above)
Comment (sahsisunny) [reply website/components/GoalForm/GoalForm.vue]: added

File: website/components/GoalForm/GoalForm.vue (hunk 4 above)
Comment (sahsisunny) [reply website/components/GoalForm/GoalForm.vue]: removed

File: website/components/GoalForm/GoalForm.vue (hunk 5 above)
Comment (sahsisunny) [reply website/components/GoalForm/GoalForm.vue]: removed

File: website/components/GoalForm/GoalForm.vue (hunk 7 above)
Comment (sahsisunny) [reply website/components/GoalForm/GoalForm.vue]: removed

File: website/components/GoalForm/GoalForm.vue (hunk 9 above)
Comment (sahsisunny) [reply website/components/GoalForm/GoalForm.vue]: removed

File: website/components/GoalForm/GoalForm.vue (hunk 8 above)
Comment (sahsisunny) [reply website/components/GoalForm/GoalForm.vue]: removed

File: website/components/GoalForm/GoalForm.vue (hunk 6 above)
Comment (sahsisunny) [reply website/components/GoalForm/GoalForm.vue]: removed

File: website/components/GoalForm/GoalForm.vue (hunk 1 above)
Comment (sahsisunny) [reply website/components/GoalForm/GoalForm.vue:63]: yes added

**Comment by sahsisunny** [comment]
Code added in another PR #102

**Pull request closed without merging**